src/test/ui/parser/default.stderr

```diff
@@ -1,30 +1,25 @@
-error: `default` is not followed by an item
-  --> $DIR/default.rs:23:5
-   |
-LL |     default pub fn foo<T: Default>() -> T { T::default() }
-   |     ^^^^^^^ the `default` qualifier
-   |
-   = note: only `fn`, `const`, `type`, or `impl` items may be prefixed by `default`
-
-error: non-item in item list
-  --> $DIR/default.rs:23:13
+error: expected one of `async`, `const`, `extern`, `fn`, `pub`, `unsafe`, or `use`, found keyword `pub`
+  --> $DIR/default.rs:24:13
    |
 LL | impl Foo for u32 {
-   |                  - item list starts here
+   |                  - while parsing this item list starting here
 LL |     default pub fn foo<T: Default>() -> T { T::default() }
-   |             ^^^ non-item starts here
-...
+   |             ^^^
+   |             |
+   |             expected one of 7 possible tokens
+   |             help: visibility `pub` must come before `default pub`: `pub default pub`
```

Comment:
Only one nitpick: this suggestion is incorrect. The span should point at `default pub` and the suggestion content should be `pub default`.

Reply:
`pub default`? Can you show me an example output you mentioned? I don't think `default pub` should be there, it should be `pub` only right?

Reply:
Actually, it should be only `default`, as [`pub` is implied](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=1d0621bc65a55ffc2cd6e712b8ac04d8), but yes the correct syntactic order is `pub default fn foo()`.

Reply:
How should I check that? It doesn't seemed to be part of `parse_fn_front_matter` but rather in `parse_fn_kind`, I think it is related to
```rust
} else if self.check_fn_front_matter() {
    // FUNCTION ITEM
    let (ident, sig, generics, body) = self.parse_fn(attrs, req_name, lo)?;
    (ident, ItemKind::Fn(def(), sig, generics, body))
```
It feels weird that `defaultness` is not parsed as part of `parse_fn_front_matter`.

I feel lazy to do this, requires cleaning on other parts, @estebank do you want to do it?

Reply:
This is fixed.